File: BackEnd/src/domain/interfaces/usecases/plants/IGetUserPlantsUseCase.ts
/**
 * @file IGetUserPlantsUseCase.ts
 * @description Interfaz del caso de uso para obtener todas las plantas del usuario.
 * @module Plants
 * @layer Domain
 */

import type { PlantResponseDTO } from '../../../dtos/plants/plant-response.dto.js';

export interface IGetUserPlantsUseCase {
  /**
   * @param userId — Id del usuario autenticado.
   * @returns Lista de plantas del usuario como DTOs de respuesta.
   */
  execute(userId: string): Promise<PlantResponseDTO[]>;
}

File: FrontEnd/lib/domain/interfaces/usecases/plants/i_get_user_plants_use_case.dart
/// @file i_get_user_plants_use_case.dart
/// @description Interfaz del caso de uso para obtener las plantas del usuario.
/// @module Plants
/// @layer Domain
library;

import '../../../entities/plant.dart';

abstract interface class IGetUserPlantsUseCase {
  /// Devuelve la lista de plantas activas del usuario autenticado.
  Future<List<Plant>> execute();
}
